### ISSUE TITLE
On some devices the calls do not work on the mobile Internet

### ISSUE THREAD

**Issue by Strikelon** [BelledonneCommunications/linphone-android#1000]
Hello! Please, give advice.
I am making an android application based on your library.
My problem is that on some devices the calls do not work on the mobile Internet, but they work good on wifi.
I have an example:
DEVICE=crosshatch
MODEL=Pixel 3 XL
MANUFACTURER=Google
SDK=29
I attach logs:
[linphone1.log](https://github.com/BelledonneCommunications/linphone-android/files/4600163/linphone1.log)
The logs say something about proxy config, But I do not understand what I need to change in the settings to solve the problem.
Thanks for the advice.

**Comment by Viish** [comment]
Hi,
I see nothing wrong in your logs except indeed at some time it fails to send packets (but later it succeeds on the same network).
I think the issue comes from the GSM network more than your device, because we do have a Pixel 3 XL and it works fine on our GSM network.
Or maybe it's on your server that the issue happens?
Cheers,

**Comment by Strikelon** [comment]
Hello. Thank you for your last reply.
I got additional information, I really hope that you can help me.

We tested calls on different mobile operators.
Test Case Example for my application:

Android- LTE calls worked on "Verizon" mobile operator.
and not worked on LTE with "T-Mobile" mobile operator..

But when we turned on VPN, calls with "T-Mobile" started to work on.

In the last answer, you said that the problem is possible in the mobile operator, but
we checked the calls on the original application linphone and the calls worked with "T-Mobile" without VPN well.

The question is, what can be configured in my application wrong? Because on the original application Linphone, calls with the T-Mobile" operator work fine.

I used "org.linphone:linphone-sdk-android:4.3+".

I attach screenshots of settings from the linphone application that we installed:
https://prnt.sc/sjgjah
https://prnt.sc/sjgjov
https://prnt.sc/sjgjzz
and the logs from linphone app with successful calls:
[linphone success.txt](https://github.com/BelledonneCommunications/linphone-android/files/4648790/linphone.success.txt)

Thank you for the opportunity to ask a question. I really hope for your help!

**Comment by Viish** [comment]
I see you are using UDP as transport protocol.
I strongly advise you against it, and it could be the source of your issue, even if it did work with our app. The fact that the VPN made it work is also a clue for that.
A lot of providers limit VoIP on their network, and a lot of routers do bad thing to UDP packets because they think they are smart. Most of the time they aren't. You can read more about that on our wiki: https://wiki.linphone.org/xwiki/wiki/public/view/Linphone/Good%20practices%20for%20using%20SIP/
Anyway try with TCP (or even better TLS) and I'm pretty sure your issue will be solved.

**Comment by Strikelon** [comment]
Hello!
UDP protocol was used on original linphone during a successful call at first, then changed to tcp and it was successful.
But in my application that I am developing based on liblinePhone, I use only tcp, but unfortunately this does not solve the problem.

Can you tell me what could be the difference that the calls work on the original application normally, but on my application there are problems with some operator?

Maybe I should try Library 4.4 version?

Thank you!

**Comment by Viish** [comment]
Are you making calls to our SDK from multiple threads at the same time? Some logs in log3.fail.log aren't "in place".
In log1.fail.log you are correctly registered but the invite isn't answer, have you checked you server logs to see if it has received it?
4.4 SDK probably won't change anything, if Linphone works fine the issue must lies in your app.

**Comment by Strikelon** [comment]
I see that many sip applications have problems with the T-Mobile operator.
https://community.t-mobile.com/message/587956#587956
I found a description of a similar problem using this link:
"After running a packet capture on my server I've found very interesting fact. While some SIP communication is passing through. Device gets registered, SIP INFO and SIP NOTIFY packets are sent, received and ACKed, but SIP INVITE (which contains information to set up the call) never makes it through. The result is timeout when I try to place a call (using native client in android. This is impossible without something selectively dropping packets, based on their payload and definitively seems to be deliberate. Only happens when using IPv4, IPv6 is unaffected and everything works properly."

As far as I remember in the linphone application, the default ipv6  support is included, maybe this is the solution, I will write the result, when I check this.
